pkg/internal/checkup/trex/client_test.go

```diff
@@ -0,0 +1,277 @@
+/*
+ * This file is part of the kiagnose project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package trex_test
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	expect "github.com/google/goexpect"
+
+	assert "github.com/stretchr/testify/require"
+
+	"github.com/kiagnose/kubevirt-dpdk-checkup/pkg/internal/checkup/trex"
+)
+
+const (
+	trafficGeneratorPacketsPerSecond = "1m"
+	testDuration                     = time.Second
+	verbosePrintsEnabled             = false
+
+	portIdx = trex.SourcePort
+)
+
+func TestGetPortStatsSuccess(t *testing.T) {
+	t.Run("when GetPortStats succeed", func(t *testing.T) {
+		expecter := stubExpecter{}
+		c := trex.NewClient(expecter, trafficGeneratorPacketsPerSecond, testDuration, verbosePrintsEnabled)
+
+		stats, err := c.GetPortStats(portIdx)
+		assert.NoError(t, err, "GetPortStats returned an error")
+		expected := trex.PortStats{
+			ID:      "razdt1qe",
+			Jsonrpc: "2.0",
+			Result: trex.PortStatsResult{
+				Ibytes:      68625,
+				Ierrors:     10,
+				Ipackets:    893,
+				MCPUUtil:    11,
+				MTotalRxBps: 6630.8935546875,
+				MTotalRxPps: 9.337542533874512,
+				MTotalTxBps: 1820482048,
+				MTotalTxPps: 3346474.25,
+				Obytes:      32640000000,
+				Oerrors:     15,
+				Opackets:    480000000,
+			},
+		}
+		assert.Equal(t, expected, stats, "GetPortStats returned unexpected result")
+	})
+}
+
+func TestGetPortStatsFailure(t *testing.T) {
+	t.Run("when fails on batch error", func(t *testing.T) {
+		expectBatchErr := errors.New("failed to run batch")
+		expecter := &stubExpecter{
+			expectBatchErr: expectBatchErr,
+		}
+
+		c := trex.NewClient(expecter, trafficGeneratorPacketsPerSecond, testDuration, verbosePrintsEnabled)
+		stats, err := c.GetPortStats(portIdx)
+
+		assert.ErrorContains(t, err, expectBatchErr.Error())
+		assert.Empty(t, stats)
+	})
+	t.Run("when fails on timeout", func(t *testing.T) {
+		expectTimeoutErr := errors.New("failed on timeout")
+		expecter := &stubExpecter{
+			timeoutErr: expectTimeoutErr,
+		}
+		c := trex.NewClient(expecter, trafficGeneratorPacketsPerSecond, testDuration, verbosePrintsEnabled)
+		stats, err := c.GetPortStats(portIdx)
+
+		assert.ErrorContains(t, err, expectTimeoutErr.Error())
+		assert.Empty(t, stats)
+	})
+}
+
+func TestGetGlobalStatsSuccess(t *testing.T) {
+	t.Run("when GetPortStats succeed", func(t *testing.T) {
```

Comment:
Please consider not using a sub-test when there is only one.

Reply:
DONE